app/controllers/projects_controller.rb

```diff
@@ -1,13 +1,61 @@
 class ProjectsController < ApplicationController
   before_action :set_project, only: [:show, :edit, :update, :destroy]
-
   http_basic_authenticate_with name: "cs169", password: ENV['PROJECTSCOPE_PASSWORD']
   
   # GET /projects
   # GET /projects.json
+
   def index
-    @projects = Project.all
+    
+    
+    if session.nil?
+      session[:order] = "ASC"
+      @projects = Project.all
+    elsif session[:pre_click].nil?
+      @projects = Project.all      
+    else
+      @metric_names = ProjectMetrics.metric_names
+      if session[:pre_click] == "project_name"
+        if session[:order] == "DESC"
+          @projects = Project.order(name: :desc)
+        else
+          @projects = Project.order(:name)
+        end
+      elsif session[:pre_click] == "code_climate" || session[:pre_click] == "github"||session[:pre_click] == "slack_trends"||session[:pre_click] == "slack"||session[:pre_click] == "pivotal_tracker"
+        if session[:order] == "DESC"
+          @projects = Project.joins(:metric_samples).where("metric_samples.metric_name = ?", session[:pre_click]).order("metric_samples.score")
+        else            
+          @projects = Project.joins(:metric_samples).where("metric_samples.metric_name = ?", session[:pre_click]).order("metric_samples.score")
+        end
+      end
+    end
+    click_type = params[:type]
+    if !click_type.nil? 
+      if session[:pre_click] == click_type
+        if session[:order] == "ASC"
+          session[:order] = "DESC"
+        else
+          session[:order] = "ASC"
+        end
+      else
+        session[:order] = "ASC"
+        session[:pre_click] = click_type
+      end
+    end
     @metric_names = ProjectMetrics.metric_names
+    if click_type == "project_name"
+      if session[:order] == "ASC"
+        @projects = Project.order(:name)
+      else
+        @projects = Project.order(name: :desc)
+      end
+    elsif click_type == "code_climate" || click_type == "github"||click_type == "slack_trends"||click_type == "slack"||click_type == "pivotal_tracker"
```

Comment:
we should avoid hard coding to this set of metrics - note you can get a list of metrics in the system using this method: https://github.com/AgileVentures/ProjectMetrics/blob/master/lib/project_metrics.rb#L9 and you can then use Array#include? to check if the click_type matches an existing metric